olog.py

```diff
@@ -46,23 +47,26 @@ def delete_logbook(self, name):
         url = f'{self._url}/resources/logbooks/{name}'
         res = self._session.delete(url, **self._kwargs)
         res.raise_for_status()
-        return res.json()
+        # Remove .json() below since it returns
+        # "JSONDecodeError: Expecting value: line 1 column 1 (char 0)"
+        return res
 
     def post_logbook(self, logbook):
         """
+        # TODO: check what it does, cannot create a logbook with the method
         Create a logbook.
         """
         url = f'{self._url}/resources/logbooks'
-        res = self._session.post(url, data=logbook)
+        res = self._session.post(url, data=logbook, **self._kwargs)
         res.raise_for_status()
         return res.json()
 
     def put_logbook(self, logbook):
         """
-        Create or update a logbook (mathced by name).
+        Create or update a logbook (matched by name).
         """
         url = f'{self._url}/resources/logbooks/{logbook["name"]}'
-        res = self._session.put(url, data=logbook)
+        res = self._session.put(url, data=json.dumps(logbook), **self._kwargs)
```

Comment:
Does `json.dumps` seem to be necessary? The [requests documentation](http://docs.python-requests.org/en/master/user/quickstart/#make-a-request) suggests a dictionary is also accepted. I would prefer to let requests handle serialization if we can.

Reply:
It didn't work otherwise, I tried a bunch of different inputs, only that one worked fine. I'd love to learn about a way to work without serialization, however the pyOlog's implementation suggests to use JSONEncode for that (not aure it's a proper solution).

Reply:
What about replacing `data=json.dumps(logbook)` with `json=logbook`?

Reply:
Yes, that worked!